### ISSUE TITLE
query with "--default"-flag fails when discovering the SQL mode

### ISSUE THREAD

**Issue by Chrico** [wp-cli/db-command#237]
## Bug Report

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).
- [x] Yes, more specifically, I reviewed the guidelines on [how to write clear bug reports](https://make.wordpress.org/cli/handbook/bug-reports/).

**Describe the current, buggy behavior**

When I run following command:

```cli
wp db query "select 1;" --defaults
```

the query fails with following message:

```cli
Error: Failed to get current SQL modes. Reason: '"mysql --no-defaults --no-auto-rehash --batch --skip-column-names --execute="SELECT' is not recognized as an internal or external command,
operable program or batch file.
```

When executing `wp db query`, than 2 SQL queries are executed internally:

1. discover SQL mode
2. execute actual query

The actual query for 2. is build first and calls `DB_Command::get_defaults_flag_string()`  [here](https://github.com/wp-cli/db-command/blob/main/src/DB_Command.php#LL450C71-L450C95). This method will remove the `$assoc_args['defaults']` [here](https://github.com/wp-cli/db-command/blob/main/src/DB_Command.php#L1923). Afterwards when `DB_Command:.get_sql_mode_query()` is called [here](https://github.com/wp-cli/db-command/blob/main/src/DB_Command.php#LL462C36-L462C54) and the `defaults` are not set anymore in `$array_assoc`. This results in adding `--no-defaults` to the query 1.

**Describe how other contributors can replicate this bug**

- Install WP-CLI 2.8.1
- Run `wp db query "SELECT 1;" --defaults`
- Interally `/usr/bin/env mysql --no-defaults /* snip */ --execute='SELECT @@SESSION.sql_mode'` is executed first before `/usr/bin/env mysql --exute='SELECT 1;'` is called.

**Describe what you would expect as the correct outcome**

The "discover SQL mode" should also not add `--no-defaults` when `wp db query {sql query} --defaults` is executed.

**Let us know what environment you are running this on**

```
OS:     Windows NT 10.0 build 22621 (Windows 10) AMD64
PHP version:    8.0.19
WP-CLI version: 2.8.1
```

**Comment by danielbachhuber** [comment]
Thanks for the report, @Chrico !

Feel free to submit a pull request, if you'd like. Here is some [guidance on our pull request best practices](https://make.wordpress.org/cli/handbook/contributions/pull-requests/).

**Comment by lkraav** [comment]
Yep since migrating to SkySQL that requires `--ssl-verify-server-cert`, this is now biting us in the butt as well. `db cli` is possible to configure with `wp-cli.yml`, but `db query` fails with configuration or cmdline because of this issue.

Exploring how to put Copilot on it and get at least a patch candidate out.

**Comment by swissspidy** [comment]
Moving this to #311
